feature/2602 - Add KSAV

jessie846 opened the pull request:
Resolves #2062 

The purpose of this pull request is to add Savannah to the airport list, and make it an airport people can control on openscope.

jessie846 committed:
added ksav

jessie846 committed:
fixed ksav airspace

jessie846 committed:
fixed spacing between aircraft

jessie846 committed:
Added video maps

enhanced the basic video map in KSAV

jessie846 committed:
Added KSAV terrain

It is blank I am struggling with the QGIS, would appreciate some help

jessie846 committed:
fixed small bug in runway numbers

jessie846 committed:
Worked on KDEN

erikquinn commented:
@jessie846 what is the DEN stuff?

erikquinn commented:
Okay, it looks like you've tossed the addition of KDEN into this pull request.

If you were to add KDEN, they should be separate pull requests, with separate corresponding issues. But as you saw on the comments to #2069, that is a duplicate of #1166 (PR #1942), so there is no point working on KDEN since the work is already done. So please remove the KDEN changes from this PR, and when ready, re-request a review from the airport-reviewers team.

jessie846 committed:
delete kden

jessie846 committed:
add ksav geo file

jessie846 committed:
Create ksav.geojson

jessie846 committed:
added a new ata to SAV,

Cleaning up Savannah and fixing any mistakes I found after play testing for around 30 minutes

jessie846 committed:
Added airport documentation for KSAV

jlambert134 committed:
Delete ksav.geojson

jlambert134 committed:
Create ksav.geojson

jlambert134 committed:
terrain for KSAV

jlambert134 committed:
Update ksav.geojson

jessie846 committed:
Update ksav.geojson

Fechulo commented on documentation/airport-guides/ksav.md:
```diff
@@ -0,0 +1,25 @@
+The SAV TRACON airspace covers about 40nm from the Savannah Airport and from the surface to 10,000 feet.
+
+## Procedures
+Savannah has no SIDS or STARS, so all aircraft enter and exit the airspace through ATA’s and DTA’s. All aircraft will exit or enter the airspace through a fix listed below. All arrivals will fly to their entry fix first and then proceed direct to the Savannah VOR (SAV). All departure will proceed direct to their first fix. 
```
No apostrophe needed in ATAs and DTAs. The 's' marks the plural, not the possessive 😉 

> All departure will ...

missing 's' 😅

jessie846 replied:
Fixed

Fechulo commented on assets/airports/ksav.json:
```diff
@@ -0,0 +1,1899 @@
+    "ctr_radius": 50,
+    "ctr_ceiling": 10000,
+    "initial_alt": 3000,
+    "position": ["N32d07m39.298", "W81d12m07.675", "10ft"],
```
Airport elevation is 50ft

jessie846 replied:
Fixed

Fechulo commented on assets/airports/ksav.json:
```diff
@@ -0,0 +1,1899 @@
+        "ZALAV":  ["N32d08m24.287", "W80d49m24.824"],
+        "ZUTAN":  ["N31d54m46.609", "W81d06m01.969"]
+    },
+    "runways":[
```
Only runways 10 and 01 appear to have ILSs

jessie846 replied:
Fixed

Fechulo commented on assets/airports/airportLoadList.json:
```diff
@@ -373,6 +373,12 @@
     },
+    {
+        "icao": "ksav",
+        "level": "easy",
```
With the currently-defined traffic levels, this falls under the 'medium' category (40-60 acph). Either reduce the traffic to be within 20-40 acph or change the level to 'medium'

jessie846 replied:
Fixed

Fechulo commented on assets/airports/ksav.json:
```diff
@@ -0,0 +1,1899 @@
+        "JANIE":  ["N31d51m33.930", "W81d17m44.602"],
+        "JROSS":  ["N32d42m40.008", "W80d37m37.990"],
+        "KUCES":  ["N32d42m40.008", "W80d37m37.990"],
+        "KSAV":   ["N32d07m39.298", "W81d12m07.675", "Savannah"],
```
This isn't needed, see my comment about the spawnPatterns

jessie846 replied:
Fixed

Fechulo commented on assets/airports/ksav.json:
```diff
@@ -0,0 +1,1899 @@
+        "TILLS":  ["N32d28m44.099", "W81d14m04.844"],
+        "TOSEA":  ["N32d14m03.037", "W81d12m03.199"],
+        "VIYAP":  ["N31d15m08.147", "W81d26m08.188"],
+        "VP334":  ["N32d02m50.000", "W81d43m58.700"],
```
The waypoints beginning with 'VP' are VFR points, so there's no reason to include them here

jessie846 replied:
Fixed

Fechulo commented on assets/airports/ksav.json:
```diff
@@ -0,0 +1,1899 @@
+            "origin": "KSAV",
+            "destination": "",
+            "category": "departure",
+            "route": "KSAV..SMALZ..CESOR",
```
You don't actually need to define KSAV as part of their route here. Alternatively you can add KSAV but make sure to include the departure rwy.

For example:
> "KSAV10..SMALZ..CESOR"

It would also be good to give departures a heading to fly after takeoff. This can be done with the commands key, which you read more about here: https://github.com/openscope/openscope/blob/develop/documentation/spawnPatternReadme.md

jessie846 replied:
Fixed

Fechulo reviewed (changes requested):
The restricted areas located inside the TRACON should be added to the airport file also (R-3005 and R-3007)

jessie846 committed:
Fixed to reflect requested changes

Added restricted airspaces, r-3005, r-3007, fix punctuation issues, updates SOP link, changed departures to Runway heading,  removed VFR points, fixed KSAV to approriate level, Fixed Instrument Approaches to reflect real runways, fixed airport elevation

jessie846 committed:
fixed runway heading

jessie846 committed:
added more realistic airlines to KSAV< changed level from medium to easy and fixed appropriate spawn rates

Fechulo commented on assets/airports/ksav.json:
```diff
@@ -0,0 +1,2124 @@
+        {
+            "name": "R-3005A",
+            "height": "29000ft",
+            "poly": [
```
Polys should be defined as a series of coordinates, not a series of lines.

Additionally, it seems some of the coordinates aren't quite right. Ideally, the restricted area should line up with the video map perfectly

![image](https://user-images.githubusercontent.com/13125311/223106317-554b52a7-2fe1-437f-ba04-007c0dbc5148.png)

![image](https://user-images.githubusercontent.com/13125311/223106332-f631e21b-e399-415d-8888-b746bc64b694.png)

Fechulo reviewed (changes requested):
Nearly there

jessie846 committed:
started working on restricted airspace
